src/datepicker/calendar-header.js

```diff
@@ -198,91 +212,94 @@ export default class CalendarHeader extends React.Component<HeaderPropsT> {
     };
   }
 
-  renderMonthDropdown() {
-    const {date, locale, overrides = {}} = this.props;
-    const [MonthSelect, monthSelectProps] = getOverrides(
-      overrides.MonthSelect,
-      Select,
-    );
-    const selectOverrides = mergeOverrides(
-      this.getSelectOverrides({width: '115px'}),
-      // $FlowFixMe
-      monthSelectProps && monthSelectProps.overrides,
-    );
-    const monthOptions = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11].map(M => ({
-      id: M,
-      label: getMonthInLocale(M, locale),
-    }));
-    return (
-      <MonthSelect
-        clearable={false}
-        value={[{id: getMonth(date)}]}
-        maxDropdownHeight={'300px'}
-        options={monthOptions}
-        searchable={false}
-        {...monthSelectProps}
-        // Adding event handlers after customers overrides in order to
-        // make sure the components functions as expected
-        // We can extract the handlers from props overrides
-        // and call it along with internal handlers by creating an inline handler
-        onChange={this.handleMonthChange}
-        // internal and incoming overrides are merged above
-        overrides={selectOverrides}
-      />
-    );
-  }
+  renderMonthYearDropdown = () => {
+    const {date, locale, maxDate, minDate, overrides = {}} = this.props;
 
-  renderYearDropdown() {
-    const {date, minDate, maxDate, overrides = {}} = this.props;
-    const [YearSelect, yearSelectProps] = getOverrides(
-      overrides.YearSelect,
+    const [MonthYearSelect, monthYearSelectProps] = getOverrides(
+      overrides.MonthYearSelect,
       Select,
     );
     const selectOverrides = mergeOverrides(
-      this.getSelectOverrides({width: '80px'}),
+      this.getSelectOverrides({width: '160px'}),
       // $FlowFixMe
-      yearSelectProps && yearSelectProps.overrides,
+      monthYearSelectProps && monthYearSelectProps.overrides,
     );
-    const maxYear = maxDate ? getYear(maxDate) : 2100;
-    const minYear = minDate ? getYear(minDate) : 1900;
-    const yearOptions = [];
+
+    const MONTHS = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
+    const maxYear = maxDate ? getYear(maxDate) : 2030;
+    const minYear = minDate ? getYear(minDate) : 1980;
```

Comment:
should we keep the original values? 1980 can be a quite limiting default, if it's used for birthdates for example

Reply:
i had considered birthdates, but thought that a calendar view is generally a poor UX for choosing them. Most sites simply have a masked input. Honestly, I tightened constraints here because it was slow to operate with all of the original values. Is a virtualized list worth pulling in here? Maybe we could roll our own virtualization to not pay as much for the bundle increase...

Reply:
that makes sense, thanks for explaining Chase 👍 

yeah, probably it's not worth doing that - let's use the updated min and max values